### PR TITLE
Update loopback version

### PR DESCRIPTION
**Pull request by jannyHou**
### Description

Update loopback versions.
5085 is using 3.x, and it needs to include the security fix in loopback@3.22.0

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by smartmouse** [comment]
we need to pass CI on node 6.x , do not care about node 4 anymore.

File: common/models/workspace.js [hunk 1]
```diff
@@ -36,14 +36,15 @@ module.exports = function(Workspace) {
     var TEMPLATE_DIR = path.join(__dirname, '..', '..', 'templates', 'projects');
     var DEFAULT_TEMPLATE = 'api-server';
     var DEPENDENCIES_3_X = {
-      'loopback': '^3.0.0',
+      'loopback': '^3.22.0',
       'loopback-component-explorer': '^4.0.0',
```
Comment (bajtos) [inline common/models/workspace.js]: Please update explorer to `^6.0.0` while you are making these changes.

File: common/models/workspace.js [hunk 2]
```diff
@@ -36,14 +36,15 @@ module.exports = function(Workspace) {
...
     };
     var DEPENDENCIES_2_X = {
-      'loopback': '^2.22.0',
-      'loopback-component-explorer': '^2.4.0',
+      'loopback': '^2.40.0',
+      'loopback-component-explorer': '^5.4.0',
```
Comment (bajtos) [inline common/models/workspace.js]: DEPENDENCIES_2_X are using versions in Active LTS mode, explorer version 5.x is the right one to use here 👍

**Comment by bajtos** [comment]
> we need to pass CI on node 6.x , do not care about node 4 anymore.

Ideally, we should drop support for Node.js 4.x in `package.json`, but that's out of scope of this pull request.

**Comment by bajtos** [comment]
> > we need to pass CI on node 6.x , do not care about node 4 anymore.

> Ideally, we should drop support for Node.js 4.x in package.json, but that's out of scope of this pull request.

`package.json` is already saying `engines: {node >=6}` in master. I realized just now that this pull request is targeting 3.x branch

**Comment by bajtos** [comment]
@jannyHou `3.x-latest` is a branch created and maintained by `slt-release` tooling. Pull request must target `3.x` branch instead.

**Comment by bajtos** [comment]
I have created the `3.x` branch from the commit 826889da, its the last meaningful change before 4.0 that is clearly not breaking anything. I have also edited the pull request to target `3.x` branch.

I am not sure what GitHub did under the hood, you may need to hard-reset your local git clone instead of doing `git pull`:

```
$ git fetch
$ git checkout update/loopback
$ git reset --hard origin/update/loopback
```

I am not sure what was the major change that caused version bump to 4.0. The only remaining change between 3.x-latest and 4.0 is this explorer-related pull request: https://github.com/strongloop/loopback-workspace/pull/496 You may want to review those changes and decide if you should apply any of them to this pull request, since you are upgrading loopback-explorer too.

I think it would be best to upgrade API Connect to use `loopback-workspace@4`. What are the arguments for staying with `loopback-workspace@3`?

**Comment by bajtos** [comment]
> I am not sure what GitHub did under the hood, you may need to hard-reset your local git clone instead of doing git pull.

PR Linter is saying that PR is out of date and needs to be rebased - maybe you don't need to hard-reset but simply `git rebase 3.x`.

**Comment by jannyHou** [comment]
@bajtos Thank you for the review and created the correct branch!

It is `generator-loopback@2.3.0`(5085 uses it) that uses `loopback-workspace@3.x`.
The minimal version `generator-loopback` using `loopback-workspace@4.x` is 5.x, which is a big jump from 2.3.0. 

I think the version bumped up from 3.x to 4.x to support a flag [`generateOperationScopedModel`](https://github.com/strongloop/loopback-workspace/pull/496/files#diff-18ca367da8eff850776d6d0288bf0e71R24) in explorer, which is related to another flag in `loopback-swagger` that enables `getUpdateOnlyProperties`.

My opinion is if 5080 scaffolds lb3/lb2 app with the old version of explorer, then that flag is not supported anyway, therefore it's fine to not include it in this PR.

**Comment by bajtos** [comment]
> I think the version bumped up from 3.x to 4.x to support a flag generateOperationScopedModel in explorer, which is related to another flag in loopback-swagger that enables getUpdateOnlyProperties.
>
> My opinion is if 5080 scaffolds lb3/lb2 app with the old version of explorer, then that flag is not supported anyway, therefore it's fine to not include it in this PR.

But isn't this pull request changing the project template that is going to be used by 5080 in such way that the new applications will start using the new version of loopback-swagger that includes `getUpdateOnlyProperties` and emit two schemas for each model ("model" and "new model")?

I am not able to fully judge the impact of the changes proposed here, I'll leave it up to you to decide.

**Pull request merged**